tiledb/sm/filter/filter_pipeline.cc

```diff
@@ -89,20 +89,83 @@ void FilterPipeline::clear() {
   filters_.clear();
 }
 
+std::tuple<Status, std::optional<std::vector<uint64_t>>>
+FilterPipeline::get_var_chunk_sizes(
+    uint32_t chunk_size, Tile* const tile, Tile* const offsets_tile) const {
+  std::vector<uint64_t> chunk_offsets;
+  if (offsets_tile != nullptr) {
+    uint64_t num_offsets =
+        offsets_tile->buffer()->size() / constants::cell_var_offset_size;
+    auto offsets = (uint64_t*)offsets_tile->buffer()->data();
+
+    uint64_t current_size = 0;
+    uint64_t min_size = chunk_size / 2;
+    uint64_t max_size = chunk_size + chunk_size / 2;
+    chunk_offsets.emplace_back(0);
+    for (uint64_t c = 0; c < num_offsets; c++) {
+      auto cell_size = c == num_offsets - 1 ?
+                           tile->buffer()->size() - offsets[c] :
+                           offsets[c + 1] - offsets[c];
+
+      // Time for a new chunk?
+      auto new_size = current_size + cell_size;
+      if (new_size > chunk_size) {
+        // Do we add this cell to this chunk?
+        if (current_size <= min_size || new_size <= max_size) {
```

Comment:
_Replying here, since github doesn't give me the option to do it on the original thread:_
Thanks for the explanation! Yeah, this is useful for eliminating such cases indeed, my concern was for a scenario of chunk size 100, cells: 50, 200, 50, 200 where your code will create 2 cells of 250, instead of 4, but I guess that is ok, better than having cells of 1 anyway.